src/screens/Profile/Notification.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { View, StyleSheet, AsyncStorage } from 'react-native';
+import { Card, Button, Paragraph } from 'react-native-paper';
+import auth from '@react-native-firebase/auth';
+import Swipeable from 'react-native-gesture-handler/Swipeable';
+import PropTypes from 'prop-types';
+
+export default class Notification extends React.Component {
+  constructor(props) {
+    super(props);
+    const { viewed } = this.props;
+    this.state = { viewed, visible: true };
+    this.onClearPress = this.onClearPress.bind(this);
+    this.onViewPress = this.onViewPress.bind(this);
+  }
+
+  componentDidMount() {
+    this.unsubscribeFromAuth = auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({ currentUserID: user.uid });
+      }
+    });
+  }
+
+  async onViewPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then((value) => {
+      const notificationArray = JSON.parse(value);
```

Comment:
This is quite a slow operation. We have the array of notifications already in the parent component, so we can probably work with that. Maybe create a handler in the parent and pass it here as a prop?